### DISCUSSION TITLE
Aligning figures

### DISCUSSION THREAD

**Discussion by Cano707** [typst/typst#1409] [Q&A] [answered]
I just started using Typst, and I am a bit confused about the alignment system.

I would like to align a figure to the right like this:
```
#align(right)[
    #figure(
        image("logo.png", width: 20%),
        caption: [wda]
    )
]
```
But the result is not aligned to the right but centered.

However, I found a workaround to accomplish this, by using the following show rule:
```
#show figure: it => align(right)[
  #it.body  
  #v(10pt, weak: true)
  #emph[#it.supplement #it.counter.display(it.numbering)]: #it.caption 
]
```
Which is cumbersome, therefore my question would be, whether I am missing something on my original attempt or if it is just not supported yet.

**Comment by laurmaedje** [comment, marked as answer]
The problem here is that the figure center-aligns itself and the inner alignment wins. :/ Sometimes you want the inner one and sometimes the outer one to win (depending on which one you control), but there is currently no good way to do that.

**Reply by Cano707** [reply]
Alright, got it. Thanks.

**Comment by KronosTheLate** [comment]
I am currently trying to have some listings (code) in figures, and I would like the code to start at the left of the page, no matter the content width. I believe I am unable to do what I want, so +1 on the need/want to be able to display figures with custom alignment.

**Reply by visika** [reply]
same

**Reply by laurmaedje** [reply]
See https://github.com/typst/typst/pull/4276. In the next release, `show figure: set align(..)` should do the trick.

**Reply by KronosTheLate** [reply]
Not only is typst already amazing, but I keep being impressed at how quickly "fires are put out", and how responsive you manage to be to random github issues xD Thank you, and good work. Typst really feels like the future of typesetting documents for me ❤️

**Reply by Dartypier** [reply]
> I am currently trying to have some listings (code) in figures, and I would like the code to start at the left of the page, no matter the content width. I believe I am unable to do what I want, so +1 on the need/want to be able to display figures with custom alignment.


For me this works better:

```
#let pseudocode(content, caption) = {
set par(justify: false)
figure(
  block(
    fill: luma(250), 
    radius: 3pt,
    stroke: .6pt + luma(200),
    inset:	(x: .45em, y: .65em),
    width: 100%,
    clip: false,
  [#align(left)[#content]]),
  caption: [#caption],
  supplement: "Codice",
  kind: "code",
)}
```

Doing `align(left)[...]` only align to the left the inner content and not the figure that remain centered.

**Reply by sam-harri** [reply]
I recently came across the same problem, wanted to have my pseudocode left aligned. I found a simple fix, just add a fractional space to your first line. like so :

```
#figure(
  kind: "algorithm",
  supplement: [Algorithm],
  
  pseudocode-list(
    booktabs: true,
    numbered-title: [funcname(arg1, arg2, arg3)],
    booktabs-stroke: 0.5pt,
    stroke: 0.2pt,
    width: 100%
  )[
    + x ← [1,2,3](start, stop, m)#h(1fr)
    + y ← map()
    + *for* i *in* x:
      + y[i] ← i
    + *return* y
  ]
)<algo:funcname>
```
